src/webui/api/torrentscontroller.cpp

```diff
@@ -124,6 +125,20 @@ namespace
             }
         }
     }
+
+    bool isValidFilePriority(const BitTorrent::FilePriority priority)
```

Comment:
I think this should be in filepriority.cpp.

Reply:
Agreed it should move to filepriority. But surely it can live inside the header file, no?

Reply:
The function will be inlined if placed in header, but it is only used in 1 place so OK.

Reply:
Coding Guidelines:
>Method definitions aren't allowed in header files

> but it is only used in 1 place so OK

This may change.

Reply:
I'm having some trouble with this. I've created a `filepriority.cpp` with the following contents, and included it in `base.pri` and the appropriate `CMakeLists.txt`

```
#include "filepriority.h"

namespace BitTorrent
{
    static bool isValidFilePriority(const BitTorrent::FilePriority priority)
    {
        switch (priority) {
        case BitTorrent::FilePriority::Ignored:
        case BitTorrent::FilePriority::Normal:
        case BitTorrent::FilePriority::High:
        case BitTorrent::FilePriority::Maximum:
        case BitTorrent::FilePriority::Mixed:
            return true;
        default:
            return false;
        }
    }
}

```

but I get this compiler error:
> webui/api/torrentscontroller.cpp:570:10: error: ‘isValidFilePriority’ is not a member of ‘BitTorrent’

Reply:
1. You should declare it in "filepriority.h" and then define in "filepriority.cpp"
2. You have not use `static` with it. `static` declares functions with **internal linkage** (i.e. it can be referred to from all scopes in the *current translation unit* only). Actually anonymous namespaces are used for the same purposes currently. But you do want to use it outside of "filepriority.cpp"! So you don't need it to be internally linked.

Reply:
Removing the `static` keyword fixed it